docs/react/use-with-create-react-app-ts.en-US.md

```diff
@@ -0,0 +1,236 @@
+
+```bash
+$ yarn run eject
+```
+
+### Import on demand
+
+[babel-plugin-import](https://github.com/ant-design/babel-plugin-import) is a babel plugin for importing components on demand ([principle](/docs/react/getting-started#Import-on-Demand)). After ejecting all config files to antd-demo, we are allowed to modify `config/webpack.config.dev.js`.
+
+Note that the following setup will effectively compile your TypeScript with ES6 module resolution and then ship it over to Babel-Loader to be transpiled.
+
+```bash
+$ yarn add babel-core babel-loader babel-plugin-import babel-plugin-import babel-preset-react babel-preset-es2015 --save-dev
+```
+
+Update the webpack loader configuration for TypeScript files.  `config/webpack.config.dev.js`
+
+```diff
+{
+  test: /\.(ts|tsx)$/,
+  include: paths.appSrc,
++  loader: 'babel!ts'
+-  loader: 'ts',
+},
+```
+
+It is recommended that you create a `.babelrc` at your project root.
+
+```diff
++{
++    "presets": ["react", "es2015"],
++    "plugins": [["import", {"libraryName": "antd", "style": "css" }]]
++}
+```
+
+Update the `tsconfig.json` file.  Windows users may find a flag change necessary for `forceConsistentCasingInFileNames`.
+```diff
+{
+  "compilerOptions": {
+    "outDir": "build/dist",
++   "module": "es6",
+-   "module": "commonjs",    
++   "target": "es6",
+-   "target": "es5",
+    "lib": ["es6", "dom"],
+    "sourceMap": true,
+    "allowJs": false,
++   "jsx": "preserve",
+-   "jsx": "react",    
+    "moduleResolution": "node",
+    "rootDir": "src",
++   "forceConsistentCasingInFileNames": false,
+-   "forceConsistentCasingInFileNames": true,
+
++   "allowSyntheticDefaultImports": true,
+
+  ...
+
+```
+
+Remove the `@import '~antd/dist/antd.css';` statement added before because `babel-plugin-import` will import styles.
+
+Then reboot `yarn start` and visit demo page, you should find that the above warning message would be gone which prove the `import on demand` config is effective now.
+
+### Customize Theme
+
+According to [Customize Theme documentation](/docs/react/customize-theme), we need `less` variables modify ability of [less-loader](https://github.com/webpack/less-loader), so we add it.
+
+```bash
+$ yarn add less less-loader --save-dev
+```
+
+```diff
+loaders: [
+  {
+    exclude: [
+      /\.html$/,
+      /\.(js|jsx)$/,
++     /\.less$/,
+      /\.css$/,
+      /\.json$/,
+      /\.svg$/
+    ],
+    loader: 'url',
+  },
+
+...
+
+  // Process JS with Babel.
+  {
+    test: /\.(js|jsx)$/,
+    include: paths.appSrc,
+    loader: 'babel',
+    query: {
+      plugins: [
+-       ['import', [{ libraryName: "antd", style: 'css' }]],
++       ['import', [{ libraryName: "antd", style: true }]],  // import less
+      ],
+   },
+
+...
+
++ // Parse less files and modify variables
++ {
++   test: /\.less$/,
++   loader: 'style!css!postcss!less?{modifyVars:{"@primary-color":"#1DA57A"}}'
++ },
+]
```

Comment:
https://github.com/ant-design/ant-design/commit/be238c788775990d4f15c9f83c16ca4db7226271

Need to update there for cra 1.x